zilliqa/src/api/eth.rs

```diff
@@ -862,7 +808,32 @@ async fn subscribe(
                 }
             }
         }
-        //"newPendingTransactions" => {},
+        SubscriptionKind::NewPendingTransactions => {
+            let full = match params {
+                pubsub::Params::None => false,
+                pubsub::Params::Bool(b) => b,
+                pubsub::Params::Logs(_) => {
+                    return Err("invalid params for newPendingTransactions".into());
+                }
+            };
+
+            if full {
+                let mut txns = node.lock().unwrap().subscribe_to_new_transactions();
```

Comment:
I'd slightly campaign for this to be called subscribe_to_new_pending_transactions() , since otherwise it's ambiguous whether these are final or pending txns (or if you need both, use an enum in the args). Don't feel strongly, but the more ambiguity we allow to creep in, the higher the cognitive load of reading the code will get...

Reply:
Yeah fair point, will update in a follow-up PR.